### PR TITLE
🐛 amp-access-poool: Fix wrong usage of global document object

### LINKED ISSUE #25665: <amp-access-poool> Cannot read property 'appendChild' of null
## What's the issue?

`<amp-access-poool>` can/should not access global `document` object for docs loaded using AMP Shadow.

## How do we reproduce the issue?

Please provide a public URL and ideally a reduced test case (e.g. on jsbin.com) that exhibits only your issue and nothing else. Provide step-by-step instructions for reproducing the issue:

https://jsfiddle.net/ft8q6pn3/1/

1. Load any `<amp-access-poool>` example using XHR
2. Attach document to shadow root using AMP Shadow

## What browsers are affected?

All of them

## Which AMP version is affected?

1911070201440

--

PR incoming

### PR DESCRIPTION
**Pull request by dackmin**
Fixes #25665  when `amp-access-poool` uses global `document` to find an element for docs loaded using AMP shadow, where the element doesn't really exist inside the DOM.

### CODE REVIEW

**Comment by dackmin** [comment]
@googlebot I consent.

**Comment by NicolasAuger** [comment]
@googlebot I consent.

File: extensions/amp-access-poool/0.1/poool-impl.js [hunk 1]
```diff
@@ -175,7 +175,7 @@ export class PooolVendor {
    * @private
    */
   renderPoool_() {
-    const pooolContainer = document.getElementById('poool');
+    const pooolContainer = this.ampdoc.win.document.getElementById('poool');
```
Comment (dvoytenko) [inline extensions/amp-access-poool/0.1/poool-impl.js]: This should actually be `this.ampdoc.getElementById(...)`

File: extensions/amp-access-poool/0.1/poool-impl.js [hunk 2]
```diff
@@ -208,10 +208,16 @@ export class PooolVendor {
    * @private
    */
   onRelease_() {
-    const articlePreview = document.querySelector('[poool-access-preview]');
+    const articlePreview = this.ampdoc.win.document.querySelector(
```
Comment (dvoytenko) [inline extensions/amp-access-poool/0.1/poool-impl.js]: `this.ampdoc.getRootNode().querySelector(...)`

File: extensions/amp-access-poool/0.1/poool-impl.js [hunk 3]
```diff
@@ -208,10 +208,16 @@ export class PooolVendor {
...
+      '[poool-access-preview]'
+    );
     articlePreview.setAttribute('amp-access-hide', '');
-    const articleContent = document.querySelector('[poool-access-content]');
+
+    const articleContent = this.ampdoc.win.document.querySelector(
```
Comment (dvoytenko) [inline extensions/amp-access-poool/0.1/poool-impl.js]: `this.ampdoc.getRootNode().querySelector(...)`

File: extensions/amp-access-poool/0.1/poool-impl.js (hunk 1 above)
Comment (dackmin) [reply extensions/amp-access-poool/0.1/poool-impl.js]: Is it better to stick with `this.ampdoc.getElementById(...)` or move to `this.ampdoc.getRootNode().querySelector('#poool)` here?

Comment (dvoytenko) [reply extensions/amp-access-poool/0.1/poool-impl.js]: I have no preference: they both execute against `rootNode` and that's what we need here.

Comment (dackmin) [reply extensions/amp-access-poool/0.1/poool-impl.js]: Alright that was my next question, thanks for the info!

**Comment by dvoytenko** [comment]
Thanks! It's approved. I'm rerunning some tests to see if failures are unrelated.

**Comment by dackmin** [comment]
yeah I noticed that too, percy and saucelabs both fail even with latest master commits

**Comment by dvoytenko** [comment]
Tests passed. Merging.

**Pull request merged**